crates/rpc-types-engine/src/payload.rs

```diff
@@ -373,6 +380,161 @@ impl ExecutionPayloadV1 {
     }
 }
 
+#[cfg(feature = "serde")]
+impl<'de> Deserialize<'de> for ExecutionPayloadV1 {
```

Comment:
I don't want to touch serde on the versioned types directly but we should implement it like this for the `ExecutionPayload` enum

deserializing all fields and then selecting the version based on the fields

Reply:
ah fair, and then i guess remove the derived deserialize from the versioned types?

Reply:
no we need proper serde on ExecutionPayloadV1 ExecutionPayloadV2 ExecutionPayloadV3

we only need manual deserde on ExecutionPayload which then is implemented like this draft